main.tf

```
@@ -20,7 +21,7 @@ resource "azurerm_monitor_diagnostic_setting" "logs" {
   count                          = length(var.resources_to_logs)
   name                           = "Diagnostic_logs"
   target_resource_id             = var.resources_to_logs[count.index]
-  log_analytics_workspace_id     = azurerm_log_analytics_workspace.this.id
+  log_analytics_workspace_id     = try(var.log_analytics_workspace_id,azurerm_log_analytics_workspace.this[0].id)
```

inutile de faire le try parce que le azurerm_log_analytics_workspace.this[0] existe seulement si var.log_analytics_workspace_id existe 

```
resource "azurerm_log_analytics_workspace" "this" {
  count               = var.log_analytics_workspace_id == null ? 1 : 0
```

```suggestion
  log_analytics_workspace_id     = var.log_analytics_workspace_id
```